### PR TITLE
thunderbird: update to 115.0

### PR DESCRIPTION
**Pull request by pascal-huber**
[ci skip]

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc
- I built this PR locally for these architectures (if supported. mark crossbuilds):
 - [x] x86_64
 - [x] armv7l (crossbuild)
 - [x] aarch64 (crossbuild)
 - [x] i686 (native)
 - [x] x86_64-musl (native)
 - [ ] ~~armv6l-musl~~ (not supported)
 - [x] aarch64-musl
 
 
 #### Notes
 - Fixed `MOZILLA_SITE`
 - The UI seems a bit buggy with switching to calendars

### CODE REVIEW

**Comment by pascal-huber** [comment]
added thunderbird-i18n (forgot to commit)

**Comment by pascal-huber** [comment]
crossbuild fails for i686

**Comment by Chocimier** [comment]
For i686 we only care about [native build in i686 masterdir](https://github.com/void-linux/void-packages#building-32bit-packages-on-x86_64).

**Comment by pascal-huber** [comment]
> For i686 we only care about [native build in i686 masterdir](https://github.com/void-linux/void-packages#building-32bit-packages-on-x86_64).

Thanks, unfortunately it also fails for the native build. I will have another look at it in the next days.

**Comment by nvkomimi** [comment]
i imagine thunderbird is going to stay in void packages, so it would make sense to update straight to 115.0 now that it's officially released and distributed

**Comment by pascal-huber** [comment]
> i imagine thunderbird is going to stay in void packages, so it would make sense to update straight to 115.0 now that it's officially released and distributed

Agree, more importantly it should also compile on i686. I haven't had much time recently because of my new job. I might find some time to rebase, update and have another go at it tomorrow.

**Comment by pascal-huber** [comment]
Updated to 115.0, not yet tested.

**Comment by pascal-huber** [comment]
Added `libotr` runtime dependency. The UI seems a bit buggy, It won't show one of my calendars and sometimes it doesn't show the main menu to switch between emails and calendars.


Some of the errors:
```
[ImapModuleLoader] Using nsImapService.cpp
JavaScript error: resource:///modules/CalStartupService.jsm, line 18: uncaught exception: ParserError: invalid line (no token ";" or ":") "Europe/Zurich[2023c]"
JavaScript error: resource:///modules/CalStartupService.jsm, line 18: NS_ERROR_XPC_JS_THREW_JS_OBJECT: ParserError: invalid line (no token ";" or ":") "Europe/Zurich[2023c]"'ParserError: invalid line (no token ";" or ":") "Europe/Zurich[2023c]"' when calling method: [calIStartupService::startup]
console.debug: "Found 0 public keys and 0 secret keys (0 protected, 0 unprotected)"
console.warn: services.settings: Failed to load last_modified.json: TypeError: NetworkError when attempting to fetch resource.
console.error: Calendar:
  Can't create calendar for <some-id> (storage, moz-storage-calendar://): TypeError: right-hand side of 'in' should be an object, got undefined
console.error: Calendar:
  Message: TypeError: zone is null
  Stack:
    convertToZone@resource:///modules/calendar/Ical.jsm:5588:26
CalDateTime.prototype.getInTimezone<@resource:///modules/CalDateTime.jsm:141:45
unwrapSetter@resource:///modules/calendar/Ical.jsm:29:22
unwrap/<@resource:///modules/calendar/Ical.jsm:25:48
now@resource:///modules/calendar/utils/calDateTimeUtils.jsm:62:17
getStartDate@chrome://calendar/content/calendar-invitations-manager.js:347:24
updateStartDate@chrome://calendar/content/calendar-invitations-manager.js:366:30
getInvitations@chrome://calendar/content/calendar-invitations-manager.js:172:10
_doInvitationsUpdate@chrome://calendar/content/calendar-invitations-manager.js:122:52
scheduleInvitationsUpdate/this.mTimer<@chrome://calendar/content/calendar-invitations-manager.js:115:18

JavaScript error: resource:///modules/calendar/Ical.jsm, line 5588: TypeError: zone is null
console.debug: "Trying to load /usr/lib/thunderbird/libotr.so"
console.debug: "Trying to load libotr.so from system's standard library locations"
console.debug: "Trying to load libotr.so.5 from system's standard library locations"
console.debug: "Successfully loaded OTR library libotr.so.5 from system's standard library locations"
JavaScript error: resource:///modules/CalCalendarManager.jsm, line 444: TypeError: right-hand side of 'in' should be an object, got undefined
JavaScript error: chrome://calendar/content/calendar-creation.js, line 646: NS_ERROR_XPC_JAVASCRIPT_ERROR_WITH_DETAILS: [JavaScript Error: "right-hand side of 'in' should be an object, got undefined" {file: "resource:///modules/CalCalendarManager.jsm" line: 444}]'[JavaScript Error: "right-hand side of 'in' should be an object, got undefined" {file: "resource:///modules/CalCalendarManager.jsm" line: 444}]' when calling method: [calICalendarManager::registerCalendar]
```

**Comment by classabbyamp** [comment]
probably relevant: https://lore.kernel.org/distributions/075a20b1-869b-78f3-9ace-e106a4d43ac4@linuxlounge.net/T/#u

**Comment by pascal-huber** [comment]
> probably relevant: https://lore.kernel.org/distributions/075a20b1-869b-78f3-9ace-e106a4d43ac4@linuxlounge.net/T/#u

indeed. alpine seems to patch icu for thunderbird: https://git.alpinelinux.org/aports/tree/main/icu/standardize-vtzone-output.patch

**Comment by classabbyamp** [comment]
rebased and tested the icu patch on x86_64-glibc. works for me

File: common/environment/setup/misc.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ GNOME_SITE="https://download.gnome.org/sources"
 KERNEL_SITE="https://www.kernel.org/pub/linux"
 CPAN_SITE="https://www.cpan.org/modules/by-module"
 PYPI_SITE="https://files.pythonhosted.org/packages/source"
-MOZILLA_SITE="https://ftp.mozilla.org/pub"
```
Comment (classabbyamp) [inline common/environment/setup/misc.sh:16]: ftp.mozilla.org still works for me. why change it? did you check that the new value works for all the other users of MOZILLA_SITE?

Comment (pascal-huber) [reply common/environment/setup/misc.sh:16]: > ftp.mozilla.org still works for me. why change it? did you check that the new value works for all the other users of MOZILLA_SITE? If this is updated it should also be updated in Manual.md

It was failing on me when I wrote this, seems to be all right again. No need to change if its up again.

**Comment by pascal-huber** [comment]
> rebased and tested the icu patch on x86_64-glibc. works for me

thanks. It also works for me on x86_64-glibc. I noticed that the icu tests fail for me, but also on master, haven't inspected any further.

File: common/environment/setup/misc.sh (hunk 1 above)
Comment (pascal-huber) [reply common/environment/setup/misc.sh]: reverted, done

**Pull request merged**